dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/Reason.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.classpath;
+
+/** The kind of linkage error against a symbol reference. */
+enum Reason {
+  /** The target class of the symbol reference is not found in the class path. */
+  CLASS_NOT_FOUND,
+
+  /**
+   * The referenced class or interface found in the class path is not compatible with the source.
+   */
+  INCOMPATIBLE_CLASS_CHANGE,
+
+  /**
+   * The target class of the symbol reference is inaccessible to the source.
```

Comment:
How is this materially different from `INACCESSIBLE_SYMBOL`? It's the same, just applied to `MethodSymbol` or `FieldSymbol`, right? Do we really need a separate enum value?

Reply:
Example difference for a method reference:
- INACCESSIBLE_CLASS for a method reference means the target class is not accessible.
- INACCESSIBLE_SYMBOL for a method reference means the method matching name and descriptor is not accessible.